calculator.rb

```diff
@@ -0,0 +1,75 @@
+
+# Program for calculator.
+puts " Welcome to the calculator "
+operators_array = %w[add + subtract - divide / multiply * % ^]
+puts operators_array
+
+# To verify whether the user enteres a valid operator
+puts "Which operation would you like to perform "
+operation = gets.chomp
+while !operators_array.include?(operation)
+  print "Invalid input,Please try again\n"
+  operation = gets.chomp
+end
+
+# To know when it needs to return an _integer_ versus a _float_.
+def convert_i_f(value)
+  if (value =~ /\./)
```

Comment:
This checks to see it if has ANY decimals in it.  So the user could enter two of them, more detection of an invalid value would be better.  

For example 3...1 is accepted